java-frontend/src/main/java/org/sonar/java/model/JVariableSymbol.java

```diff
@@ -37,4 +43,78 @@ public VariableTree declaration() {
     return (VariableTree) super.declaration();
   }
 
+
+  static class ParameterPlaceholderSymbol extends Symbols.DefaultSymbol implements Symbol.VariableSymbol {
```

Comment:
We should fix the 3 locations in org.sonar.java.model.JUtils that always cast a `VariableSymbol` using `(JVariableSymbol)`. It's not anymore the case, it can be for example:
```
  public static boolean isEffectivelyFinal(Symbol.VariableSymbol variableSymbol) {
    return (variableSymbol instanceof JVariableSymbol) && ((IVariableBinding) ((JVariableSymbol) variableSymbol).binding).isEffectivelyFinal();
  }
```